pkg/server/server.go

```diff
@@ -159,10 +160,19 @@ func (c *Server) Run(stopCh <-chan struct{}) {
 		http.ListenAndServe(":21363", &healthzHandler{})
 	}()
 	if err := c.httpServer.Serve(c.listener); err != nil {
-		logrus.WithError(err).Fatal("http server exited")
+		logrus.WithError(err).Warning("http server exited")
 	}
 }
 
+func (c *Server) Close() {
```

Comment:
adding a new method to try and clean up goroutines that leak.

Reply:
This is good, and I think we should call this when the stopCh which is passed to `Run` is closed.  That way `Close` is called on SIGTERM and SIGINT ([[1]](https://github.com/kubernetes-sigs/aws-iam-authenticator/blob/ca0331a3b80448bf432aec17bbce7c1e730c4b10/cmd/aws-iam-authenticator/server.go#L54), [[2]](https://pkg.go.dev/sigs.k8s.io/controller-runtime/pkg/manager/signals#SetupSignalHandler)).

Reply:
@nckturner Could we do this in a follow please?

Reply:
Let's do this in a follow up